handlers/filter_output.go

```diff
@@ -0,0 +1,181 @@
+	if ver.Version != 1 {
+		initialVersion, err = dc.GetVersion(ctx, userAccessToken, "", "", collectionID, datasetID, edition, "1")
+		if err != nil {
+			setStatusCode(req, w, err)
+			return
+		}
+		initialVersionReleaseDate = initialVersion.ReleaseDate
+	}
+
+	dims := dataset.VersionDimensions{Items: nil}
+
+	filterOutput := filter.Model{}
+
+	getDimensionOptions := func(dim filter.ModelDimension) ([]string, error) {
+		q := dataset.QueryParams{Offset: 0, Limit: 1000}
+		opts, err := dc.GetOptions(ctx, userAccessToken, "", collectionID, datasetModel.ID, edition, strconv.Itoa(ver.Version), dim.Name, &q)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get options for dimension: %w", err)
+		}
+
+		var options []string
+		for _, opt := range opts.Items {
+			options = append(options, opt.Label)
+		}
+
+		return options, nil
+	}
+
+	getAreaOptions := func(dim filter.ModelDimension) ([]string, error) {
+		areas, err := dimsc.GetAreas(ctx, dimension.GetAreasInput{
+			UserAuthToken: userAccessToken,
+			DatasetID:     filterOutput.PopulationType,
+			AreaTypeID:    dim.ID,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed to get dimension areas: %w", err)
+		}
+
+		var options []string
+		for _, area := range areas.Areas {
+			options = append(options, area.Label)
+		}
+
+		return options, nil
+	}
+
+	getOptions := func(dim filter.ModelDimension) ([]string, error) {
+		if dim.IsAreaType != nil && *dim.IsAreaType {
+			return getAreaOptions(dim)
+		}
+
+		return getDimensionOptions(dim)
+	}
+
+	filterOutput, err = fc.GetOutput(ctx, userAccessToken, "", "", collectionID, filterOutputID)
+	if err != nil {
+		log.Error(ctx, "failed to get filter output", err, log.Data{"filter_output_id": filterOutputID})
+		setStatusCode(req, w, err)
+		return
+	}
+
+	for i, dim := range filterOutput.Dimensions {
+		options, err := getOptions(dim)
+		if err != nil {
+			log.Error(ctx, "failed to get options for dimension", err, log.Data{"dimension_name": dim.Name})
+			setStatusCode(req, w, err)
+			return
+		}
+
+		dim.Options = options
+		filterOutput.Dimensions[i] = dim
+
+	}
+
+	if filterOutput.Downloads == nil {
+		log.Warn(ctx, "filter output downloads are nil", log.Data{"filter_output_id": filterOutputID})
+		filterOutput.Downloads = make(map[string]filter.Download)
+	}
+
+	if form == "get-data" && format == "" {
+		isValidationError = true
+	}
+
+	if form == "get-data" && format != "" {
+		for ext, download := range filterOutput.Downloads {
+			if strings.EqualFold(ext, format) {
+				http.Redirect(w, req, download.URL, http.StatusFound)
+			}
+		}
+	}
+
+	basePage := rend.NewBasePageModel()
+	m := mapper.CreateCensusDatasetLandingPage(ctx, req, basePage, datasetModel, ver, []dataset.Options{}, dims, initialVersionReleaseDate, hasOtherVersions, allVers.Items, latestVersionNumber, latestVersionURL, lang, numOptsSummary, isValidationError, true, filterOutput)
```

Comment:
For function calls with so many arguments you may find it's easier to read if you stack the arguments vertically